xarray/core/dataset.py

```diff
@@ -1475,14 +1475,14 @@ def __getitem__(self, key):
         else:
             return self._copy_listed(np.asarray(key))
 
-    def __setitem__(self, key: Hashable, value) -> None:
-        """Add an array to this dataset.
+    def __setitem__(self, key, value) -> None:
+        """Add a (list of) array to this dataset.
```

Comment:
I think this could be clearer — check out other methods that take a single item or a list

Reply:
Ok this should look better now :)